### PR TITLE
Fix offline trace files feature

### PR DESCRIPTION
**Pull request by nordominus**
The feature OfflineTrace was working before commit ac44c47265528ddd29afbb9a72bcf095a877ecd,
but with this commit `dlt_daemon_configuration_load()` was moved from within `dlt_daemon_init()` to new `dlt_daemon_load_runtime_configuration()`.
It is called  later in` dlt_daemon_init()` in line dlt-daemon.c:788, which is too late to for reaching `dlt_offline_trace_init()` called by `dlt_daemon_local_init_p2()` called in line dlt-daemon.c:713.

The following configuration was used for verifying the issue and proposed fix

_Configure dlt-runtime.cfg_:
	LoggingMode=3

_Configure dlt.conf_:
	OfflineTraceDirectory = /tmp/dltlogs
	OfflineTraceFileSize = 100
	OfflineTraceMaxSize = 1000
	OfflineTraceFileNameTimestampBased = 0

The program was tested solely for our own use cases, which might differ from yours.

Andreas Seidl andreas.seidl@daimler.com, Mercedes-Benz AG on behalf of MBition GmbH
https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md
Licensed under MPL-2.0

### CODE REVIEW

File: src/daemon/dlt-daemon.c [hunk 1]
```diff
@@ -1287,13 +1305,8 @@ void dlt_daemon_local_cleanup(DltDaemon *daemon, DltDaemonLocal *daemon_local, i
     /* Ignore result */
     dlt_file_free(&(daemon_local->file), daemon_local->flags.vflag);
 
-#ifndef DLT_USE_UNIX_SOCKET_IPC
```
Comment (ssugiura) [inline src/daemon/dlt-daemon.c]: This change should go to separate commit, as you are reverting [https://github.com/GENIVI/dlt-daemon/commit/7d9d156fb8571aff4eb89db201a26af0d0370dc7](url).

File: src/daemon/dlt-daemon.c [hunk 2]
```diff
@@ -957,6 +949,32 @@ int dlt_daemon_local_init_p2(DltDaemon *daemon, DltDaemonLocal *daemon_local, in
...
+                                   daemon_local->flags.offlineTraceFilenameTimestampBased) == -1) {
+            dlt_log(LOG_ERR, "Could not initialize offline trace\n");
+            return -1;
+        }
+    }
+
+    return 0;
+}
+
+
```
Comment (ssugiura) [inline src/daemon/dlt-daemon.c]: Extra line. I propose to run uncrustify to your modified code, as it's written in README.

**Comment by nordominus** [comment]
Hi @gunnarx and @ssugiura quick update on the current status:
We plan to have a solution ready next week for the legal topics, so i can rework the pull requests accordingly. Many thanks for your patience!

**Comment by ssugiura** [comment]
@nordominus Thank you for your update. It's great to hear that. Hope to hear from you soon!

**Comment by ssugiura** [comment]
@nordominus Friendly reminder. Also, please resolve conflicts to finalize this commit.

File: src/daemon/dlt-daemon.c (hunk 1 above)
Comment (nordominus) [reply src/daemon/dlt-daemon.c]: I cannot find this commit anymore. Was there a force-push?

Comment (ssugiura) [reply src/daemon/dlt-daemon.c]: This is still available. If you cannot jump from the link, please try to find "7d9d156 dlt-daemon: unlink application socket" from the master branch.

**Comment by nordominus** [comment]
Hi @ssugiura 
i think this issue is fixed now by Duplicate of #175, isn't it?

**Comment by ssugiura** [comment]
@nordominus Yes, you are right. I will close this PR.
